src/Collections-Stack/NewStack.class.st

```diff
@@ -0,0 +1,97 @@
+"
+I am a parametrizable stack. That is to say, I provide the interface of a stack but depending on my #strategy, I can use a LinkedList, an OrderedCollection, etc for implementation details.
+"
+Class {
+	#name : #NewStack,
+	#superclass : #Collection,
+	#instVars : [
+		'strategy'
+	],
+	#category : #'Collections-Stack-Base'
+}
+
+{ #category : #factory }
+NewStack class >> usingLinkedList [
+ 	"Returns an instance of Stack using a LinkedList under the hood."
+ 	^ self usingStrategy: LinkedListStackStrategy new
+]
+
+{ #category : #factory }
+NewStack class >> usingOrderedCollection [
+ 	"Returns an instance of Stack using a OrderedCollection under the hood."
+ 	^ self usingStrategy: OrderedCollectionStackStrategy new
+]
+
+{ #category : #'instance creation' }
+NewStack class >> usingStrategy: aStackStrategy [
+ 	"Given a aStackStrategy, returns an instance of Stack using this strategy under the hood."
+ 	^ self new
+ 		initializeWith: aStackStrategy;
+ 		yourself
+]
+
+{ #category : #adding }
+NewStack >> add: newObject [
+ 	"Include newObject as one of the receiver's elements. Answer newObject. 
+ 	ArrayedCollections cannot respond to this message."
+
+  	^ self push: newObject
+]
+
+{ #category : #enumerating }
+NewStack >> do: aBlock [
+ 	"Evaluate aBlock with each of the receiver's elements as the argument."
+
+  	self strategy do: aBlock
+]
+
+{ #category : #initialization }
+NewStack >> initialize [
+	super initialize.
+	strategy := LinkedListStackStrategy new
```

Comment:
Since the original stack was inheriting from LinkedList, I thought that it was a good idea to make the default strategy a linked list. But we can change that if needed.